### ISSUE TITLE
renaming binary asset causes content to be corrupted

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#1501]
1. add png to assets
 2. rename the asset
 3. chrome complains that plugin (application/octet-stream) is unsupported

I saw something similar when trying to fork a notebook with binary assets for #1470 - I worked around it there by using the github-raw (base64) content instead.

**Comment by s-u** [comment]
The problem appears to be on the incoming side to JS - i.e. the notebook is already corrupted when it arrives on the JS side. Since renaming is apparently a client-side get+append+delete operation the corrupted content is then propagated back to the gist.

**Comment by s-u** [comment]
This is a bug in rserve-js, in particular this line:
```js
            var a = s.make(Uint8Array).subarray(0, l2).buffer;
```
returns the whole buffer of the message, not the actual payload of the raw vector. It seems that a reasonably portable fix is
```js
            var a = new Uint8Array(s.make(Uint8Array).subarray(0, l2)).buffer;
```
fixed accordingly upstream att/rserve-js@f921baf0
